code/main/media/mediaModule.brs

```diff
@@ -0,0 +1,245 @@
+' ********************** Copyright 2023 Adobe. All rights reserved. **********************
+' *
+' * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+' * you may not use this file except in compliance with the License. You may obtain a copy
+' * of the License at http://www.apache.org/licenses/LICENSE-2.0
+' *
+' * Unless required by applicable law or agreed to in writing, software distributed under
+' * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+' * OF ANY KIND, either express or implied. See the License for the specific language
+' * governing permissions and limitations under the License.
+' *
+' *****************************************************************************************
+
+' *********************************** MODULE: Media ***************************************
+function _adb_isMediaModule(module as object) as boolean
+    return (module <> invalid and module.type = "com.adobe.module.media")
+end function
+
+function _adb_MediaModule(configurationModule as object, identityModule as object) as object
+    if _adb_isConfigurationModule(configurationModule) = false then
+        _adb_logError("_adb_MediaModule() - configurationModule is not valid.")
+        return invalid
+    end if
+
+    if _adb_isIdentityModule(identityModule) = false then
+        _adb_logError("_adb_EdgeModule() - identityModule is not valid.")
+        return invalid
+    end if
+
+    module = _adb_AdobeObject("com.adobe.module.media")
+
+    module.Append({
+        ' constants
+        _CONSTANTS: _adb_InternalConstants(),
+
+        ' external dependencies
+        _configurationModule: configurationModule,
+        _identityModule: identityModule,
+        _edgeRequestWorker: _adb_EdgeRequestWorker(),
+        _sessionManager: _adb_MediaSessionManager(),
+
+        ' Event data example:
+        ' {
+        '     clientSessionId: "xx-xxxx-xxxx",
+        '     tsObject: { ts_inISO8601: "2019-01-01T00:00:00.000Z", ts_inMillis: 1546300800000 },
+        '     xdmData: { xdm: {} }
```

Comment:
nit: xdmData: { xdm: { eventType: "...", ...} }
This will guarantee that we will always have eventType and don't add additional checks for it.